tiledb/sm/array_schema/array_schema_evolution.cc

```diff
@@ -152,9 +152,12 @@ shared_ptr<ArraySchema> ArraySchemaEvolution::evolve_schema(
   for (auto& attr_name : attributes_to_drop_) {
     bool has_attr = false;
     throw_if_not_ok(schema->has_attribute(attr_name, &has_attr));
-    if (has_attr) {
-      throw_if_not_ok(schema->drop_attribute(attr_name));
+    if (!has_attr) {
```

Comment:
This test is not quite right. The issue is with capnp deserialization. As with all input that comes in from the outside, here from the network, we need to validate inputs. We may not assume that a capnp message was generated by `libtiledb` necessarily; it might have been formed by other, likely malicious, software. We don't need to care why it was created, only that it's detected and rejected.

There's a check in `drop_attribute` to see if the attribute is the name of an attribute that was previously added. In this case, there's no check to see if the dropped attribute here also appears in the list of added attributes.